src/Microsoft.Health.Fhir.SqlServer/Features/Storage/TvpRowGeneration/Merge/DateTimeSearchParamListRowGenerator.cs

```diff
@@ -0,0 +1,50 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.SqlServer.Features.Schema.Model;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Storage.TvpRowGeneration
+{
+    internal class DateTimeSearchParamListRowGenerator : MergeSearchParameterRowGenerator<DateTimeSearchValue, DateTimeSearchParamListRow>
+    {
+        private short _lastUpdatedSearchParamId;
+
+        public DateTimeSearchParamListRowGenerator(SqlServerFhirModel model, SearchParameterToSearchValueTypeMap searchParameterTypeMap)
+            : base(model, searchParameterTypeMap)
+        {
+        }
+
+        internal override bool TryGenerateRow(short resourceTypeId, long resourceSurrogateId, short searchParamId, DateTimeSearchValue searchValue, HashSet<DateTimeSearchParamListRow> results, out DateTimeSearchParamListRow row)
```

Comment:
Should we add a check if searchValue is null?

Reply:
This is the behavior of the previous code.